lib/taiko.js

```diff
@@ -345,33 +359,53 @@ module.exports.client = () => clientProxy;
  * Allows switching between tabs using URL or page title.
  *
  * @example
- * await switchTo('https://taiko.gauge.org/') # switch using URL
- * await switchTo('Taiko') # switch using Title
+ * # switch using URL
+ * await switchTo('https://taiko.gauge.org/')
+ * # switch using Title
+ * await switchTo('Taiko')
+ * # switch using regex URL
+ * await switchTo(/http(s?):\/\/(www?).google.(com|co.in|co.uk)/)
+ * # switch using regex Title
+ * await switchTo(/Go*gle/)
  *
  * @param {string} targetUrl - URL/Page title of the tab to switch.
  *
  * @returns {Promise}
  */
+
 module.exports.switchTo = async targetUrl => {
   validate();
   let t = typeof targetUrl;
-  if (t !== 'string') {
+  let q = Object.prototype.toString
+    .call(targetUrl)
+    .includes('RegExp');
+  let isRegexText = '';
+  if (t != 'string' && !q) {
     throw new TypeError(
-      'The "targetUrl" argument must be of type string. Received type ' +
+      'The "targetUrl" argument must be of type string or regex. Received type ' +
         t,
     );
   }
+  if (q) {
+    targetUrl = new RegExp(targetUrl);
+    isRegexText = ' Regex';
+  }
   const targets = await targetHandler.getCriTargets(
     targetUrl,
     currentHost,
     currentPort,
   );
   if (targets.matching.length === 0) {
-    throw new Error('No target with given URL/Title found.');
+    throw new Error(
+      `No target with given ${isRegexText} URL/Title found.`,
```

Comment:
This can be a generic message to avoid the conditions on `isRegexText` 
`No tab(s) matching ${tartgetURL} found`